### ISSUE TITLE
Can't use IpAddress as table index

### ISSUE THREAD

**Issue by haxtibal** [pief/python-netsnmpagent#24]
Using an IpAddress object as table index fails with
```
AttributeError: 'IpAddress' object has no attribute '_asntype'
```
Example snippet for how to reproduce:
```python
tableWithIpIndex = agent.Table(
	oidstr = "SIMPLE-MIB::thirdTable",
	indexes = [
		agent.IpAddress()
	],
	columns = [
		(2, agent.DisplayString("")),
		(3, agent.IpAddress("0.0.0.0"))
	],
)
row1 = tableWithIpIndex.addRow([agent.IpAddress("192.168.0.1")])
row1.setRowCell(2, agent.DisplayString("Some value"))
row1.setRowCell(3, agent.IpAddress("192.168.0.1"))
````

The first part of a potential fix is simple, just initialize IpAddress with the missing attributes, e.g.:
````python
class IpAddress(object):
	def __init__(self):
		self._flags = WATCHER_FIXED_SIZE
		self._asntype = ASN_IPADDRESS
		self._cvar = ctypes.c_uint(0)
		self._data_size = ctypes.sizeof(self._cvar)
		self._max_size  = self._data_size
		self.update(initval)
````

Now comes the tricky part. There's an [open Net-SNMP Bug 2136](https://sourceforge.net/p/net-snmp/bugs/2136/). Its impact is that Net-SNMP expects an IP address in network byte order if used as column instance, but in host byte order if used as index. This hits us if we're on little-endian machines like Intel x86-64.

Without further measures our example would give the correct cell value, but a reversed index
```
AGENTX-MIB::agentxObjects.30187.1.2.6.1.3.1.0.168.192 = IpAddress: 192.168.0.1
```

I'll send a pull request soon, with a suggestion how to fix the second issue too. It will produce expected output (at least if RFC 3416, section 4.2.2.1 is taken as a reference for how it should be)
````
AGENTX-MIB::agentxObjects.30187.1.2.6.1.3.192.168.0.1 = IpAddress: 192.168.0.1
````

**Comment by pief** [comment]
Excellent report!

So you're saying that net-snmp has an open bug that might get fixed someday so that we have to take care of detecting what byte order they expect the index in?

**Comment by haxtibal** [comment]
You're right, and it would be a comprehensible reason for not taking action in python-netsnmpagent. Instead, wait for Net-SNMP to get their bug fixed and tell python-netsnmpagent users "IpAddress index works only with Net-SNMP >= <what.ever.version>".

On the other hand, the Net-SNMP bug is open for more than 6 years now, although the bug submitter did provide a patch. If the Net-SNMP guys really fixed the bug, they break existing agents that depend on the buggy behaviour. So I think it's doubtful if the bug will ever get fixed.

Do you want to progress with this issue here? If yes, I could try asking at the netsnmp-coders-list about the state of #2136 first. Or just go ahead with a PR as starting point for a review. What would you prefer?

**Comment by pief** [comment]
I'd say if the PR is not too much work, go ahead with it. Always more important to provide a fix for python-netsnmpagent right now, we can always catch up with eventual net-snmp progress later.

**Comment by pief** [comment]
Fixed through PR #25 and 519b0e8.
